### PR TITLE
test: change the way credentials are loaded

### PR DESCRIPTION
**Pull request by jiaxunsongucb**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Change the way credentials are loaded to make it work for Azure Stack test pipelines.
Modify docs accordingly.
Use -mod=vendor to build binaries.
Update blobfuse version for testing.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by jiaxunsongucb** [comment]
The sanity and integration test failures are probably caused by the blobfuse update. How can I make sure I am using the same version as the docker image is using?

**Comment by andyzhangx** [comment]
> The sanity and integration test failures are probably caused by the blobfuse update. How can I make sure I am using the same version as the docker image is using?

current 1.3.5 blobfuse should also work, otherwise need update current CI tests.

**Comment by andyzhangx** [comment]
> > The sanity and integration test failures are probably caused by the blobfuse update. How can I make sure I am using the same version as the docker image is using?
> 
> current 1.3.5 blobfuse should also work, otherwise need update current CI tests.

you may build standalone blobfuse binary in this case

**Comment by jiaxunsongucb** [comment]
> > The sanity and integration test failures are probably caused by the blobfuse update. How can I make sure I am using the same version as the docker image is using?
> 
> current 1.3.5 blobfuse should also work, otherwise need update current CI tests.

I am using 1.3.5 in this PR. Let me check and look for alternative solutions later. Thanks!

**Comment by andyzhangx** [comment]
try build from source: https://github.com/Azure/azure-storage-fuse/wiki/1.-Installation#option-2---build-from-source

File: test/utils/credentials/credentials_test.go [hunk 1]
```diff
@@ -68,21 +46,34 @@ func TestCreateAzureCredentialFileOnAzurePublicCloud(t *testing.T) {
 		os.Setenv(aadClientSecretEnvVar, "")
 		os.Setenv(resourceGroupEnvVar, "test-resource-group")
 		os.Setenv(locationEnvVar, "test-location")
-		withAzureCredentials(t, false)
+		withAzureCredentials(t)
+	})
+
+	t.Run("WithEnvironmentVariables", func(t *testing.T) {
+		os.Setenv(tenantIDEnvVar, "test-tenant-id")
```
Comment (andyzhangx) [inline test/utils/credentials/credentials_test.go:53]: define these strings as const

File: pkg/blobplugin/Dockerfile [hunk 2]
```diff
@@ -18,7 +18,7 @@ ADD . .
 ARG TARGETARCH
 ARG TARGETOS
 ARG LDFLAGS
-RUN CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH go build -a -ldflags "${LDFLAGS:--X sigs.k8s.io/blob-csi-driver/pkg/blob.driverVersion=latest}" -o _output/blobplugin ./pkg/blobplugin
+RUN CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH go build -a -ldflags "${LDFLAGS:--X sigs.k8s.io/blob-csi-driver/pkg/blob.driverVersion=latest}" -mod vendor -o _output/blobplugin ./pkg/blobplugin
```
Comment (andyzhangx) [inline pkg/blobplugin/Dockerfile:21]: it's better create another PR to support `-mod vendor` first, that would be quick to merge

**Comment by jiaxunsongucb** [comment]
> try build from source: https://github.com/Azure/azure-storage-fuse/wiki/1.-Installation#option-2---build-from-source

I tried building from source, apt install, neither works.
I tried to follow the Dockerfile to install blobfuse latest version, does not work either.
But all the above approaches work on Azure Stack CI.
I will create another PR to update the blobfuse version.
@andyzhangx I think we need to make sure the image `gcr.io/k8s-testimages/kubekins-e2e:v20201111-a263fd7-master` prow job is using is based on `k8s.gcr.io/build-image/debian-base-amd64:v2.1.3`, which is the base in Dockerfile https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/pkg/blobplugin/Dockerfile.

File: test/e2e/README.md [hunk 3]
```diff
@@ -2,9 +2,33 @@
 
 ## Run E2E tests Locally
 ### Prerequisite
- - Make sure a kubernetes cluster(with version >= 1.13) is set up and kubeconfig is under `$HOME/.kube/config`
- - Copy out `/etc/kubernetes/azure.json` under one agent node to local machine
- > For AKS cluster, need to modify `resourceGroup` to the node resource group name inside `/etc/kubernetes/azure.json`
+ - Make sure a kubernetes cluster(with version >= 1.13) is set up and kubeconfig is under `$HOME/.kube/config`. Make sure kubectl is functional.
+ - Better to do: install Helm and Tiller
```
Comment (andyzhangx) [inline test/e2e/README.md:6]: `Tiller` is not needed from helm v3

Comment (andyzhangx) [reply test/e2e/README.md:6]: pls update with helm v3 doc

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**